modules/peers.js

```diff
@@ -421,33 +421,40 @@ Peers.prototype.acceptable = function (peers) {
 
 /**
  * Gets peers list and calculated consensus.
- * @param {Object} options - Constains limit, broadhash.
+ * @param {Object} options
+ * @param {number} options.limit[=constants.maxPeers] - maximum number of peers to get
+ * @param {string} options.broadhash[=null] - broadhash to match peers by
+ * @param {Array} options.allowedStates[=[2]] - allowed peer states
+ * @param {number} options.attempt[=undefined] - if 0: return peers with equal options.broadhash
+ *                                               if 1: return peers with different options.broadhash
+ *                                               if not specified: return peers regardless of options.broadhash
  * @param {function} cb - Callback function.
  * @returns {setImmediateCallback} error | peers, consensus
  */
 Peers.prototype.list = function (options, cb) {
```

Comment:
That function was sad before changes, and now is even more sad. It should be slightly refactored to make it more flexible and simplify the logic inside. So, we don't need things like `attemptsDescriptions`, `attempts`, instead we can pass parameters like `matchedBroadhashOnly`/`unmatchedBroadhashOnly` and `calculateConsensus`. Also that function shouldn't modify defined input params. Code comments are no longer valid.

Reply:
I will solve the issue by backporting calculating consensus in the separate place. List function can be refactored as a part of separate PR.

Reply:
Can you refactor that as part of this PR? That function is really hard to read.